collectors/node/cpu.go

```diff
@@ -148,22 +148,25 @@ func calculatePcts(lastTimes cpu.TimesStat, curTimes cpu.TimesStat) cpu.TimesSta
 		totalDelta = 1 // can't divide by zero
 	}
 	return cpu.TimesStat{
-		User:      round(math.Dim(curTimes.User, lastTimes.User) / totalDelta * 100),
-		System:    round(math.Dim(curTimes.System, lastTimes.System) / totalDelta * 100),
-		Idle:      round(math.Dim(curTimes.Idle, lastTimes.Idle) / totalDelta * 100),
-		Nice:      round(math.Dim(curTimes.Nice, lastTimes.Nice) / totalDelta * 100),
-		Iowait:    round(math.Dim(curTimes.Iowait, lastTimes.Iowait) / totalDelta * 100),
-		Irq:       round(math.Dim(curTimes.Irq, lastTimes.Irq) / totalDelta * 100),
-		Softirq:   round(math.Dim(curTimes.Softirq, lastTimes.Softirq) / totalDelta * 100),
-		Steal:     round(math.Dim(curTimes.Steal, lastTimes.Steal) / totalDelta * 100),
-		Guest:     round(math.Dim(curTimes.Guest, lastTimes.Guest) / totalDelta * 100),
-		GuestNice: round(math.Dim(curTimes.GuestNice, lastTimes.GuestNice) / totalDelta * 100),
-		Stolen:    round(math.Dim(curTimes.Stolen, lastTimes.Stolen) / totalDelta * 100),
+		User:      roundAboveZero((curTimes.User - lastTimes.User) / totalDelta * 100),
+		System:    roundAboveZero((curTimes.System - lastTimes.System) / totalDelta * 100),
+		Idle:      roundAboveZero((curTimes.Idle - lastTimes.Idle) / totalDelta * 100),
+		Nice:      roundAboveZero((curTimes.Nice - lastTimes.Nice) / totalDelta * 100),
+		Iowait:    roundAboveZero((curTimes.Iowait - lastTimes.Iowait) / totalDelta * 100),
+		Irq:       roundAboveZero((curTimes.Irq - lastTimes.Irq) / totalDelta * 100),
+		Softirq:   roundAboveZero((curTimes.Softirq - lastTimes.Softirq) / totalDelta * 100),
+		Steal:     roundAboveZero((curTimes.Steal - lastTimes.Steal) / totalDelta * 100),
+		Guest:     roundAboveZero((curTimes.Guest - lastTimes.Guest) / totalDelta * 100),
+		GuestNice: roundAboveZero((curTimes.GuestNice - lastTimes.GuestNice) / totalDelta * 100),
+		Stolen:    roundAboveZero((curTimes.Stolen - lastTimes.Stolen) / totalDelta * 100),
 	}
 }
 
-// Helper function for rounding to two decimal places
-func round(f float64) float64 {
+// Helper function for rounding to two decimal places. Coerces negative numbers to zero.
+func roundAboveZero(f float64) float64 {
```

Comment:
sorry to be a pain
but name `roundAboveZero` suggests f > 0, in our case it can be 0, right?